force_wfmanager/left_side_pane/new_entity_modal.py

```diff
@@ -153,3 +190,126 @@ def get_plugin_from_factory(self, factory):
             plugin = factory.mco_factory.plugin
 
         return plugin
+
+    def _get_current_model_editable(self):
+        """A check which indicates if 1. A view with at least
+        one item exists for this model and 2. That those items
+        are actually visible to the user"""
+        if self.current_model is None:
+            return False
+
+        view_info = self.view_structure()
+
+        # Remove any non-visible traits
+        view_info = [trait_name for trait_name in
+                     view_info if trait_name in
+                     self.current_model.visible_traits()]
+
+        if view_info == []:
+            return False
+
+        return True
+
+    def _get_model_description_HTML(self):
+        """Format a description of the currently selected model and it's
+        parameters, using desc metadata from the traits in
+        ${model_name}_model.py"""
+
+        # A default message when no model selected
+        if self.selected_factory is None or self.current_model is None:
+            return html_string.format("No Model Selected", "", "")
+
+        model_name = self.selected_factory.get_name()
+        model_desc = self.selected_factory.get_description()
+        view_info = self.view_structure()
+
+        # Message for a model without editable traits
+        if view_info == []:
+            return html_string.format(model_name, model_desc, "")
+
+        # Remove traits in the view which are not editable
+        view_info = [trait_name for trait_name in
+                     view_info if trait_name in
+                     self.current_model.visible_traits()]
+
+        # Retrieve descriptions from trait metadata
+        for i, trait_name in enumerate(view_info):
+            trait = self.current_model.trait(trait_name)
+            trait_desc = trait.desc
+
+            if trait_desc is not None:
+                view_info[i] = [trait_name, trait_desc]
```

Comment:
but then you change it to a list of lists, and modify it while iterating on it. I suggest using a different list.